lib/core/CStateMachine.cc

```diff
@@ -201,14 +210,15 @@ void CStateMachine::CMachineDeque::capacity(std::size_t capacity) {
     m_Capacity = capacity;
 }
 
-const CStateMachine::SMachine& CStateMachine::CMachineDeque::operator[](std::size_t pos) const {
+const CStateMachine::SMachine& CStateMachine::CMachineDeque::operator[](std::size_t pos_) const {
+    std::size_t pos{pos_};
     for (const auto& machines : m_Machines) {
         if (pos < machines.size()) {
             return machines[pos];
         }
         pos -= machines.size();
     }
-    LOG_ABORT(<< "Invalid index '" << pos << "'");
+    LOG_ABORT(<< "Invalid index '" << pos_ << "'");
```

Comment:
something for a follow up? The abort is a drastic decision as in this issue we effectively kill the process although only a feature broke. We are more forgiving in other areas, e.g. restore returns a bool.

Reply:
Yes. In hindsight, this is better dealt with by trapping and reinitialising the offending object (which would have worked around this problem). I'll make that change in a follow up PR.